webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/apidata/JsonResponseObjects.scala

```diff
@@ -1298,21 +1400,17 @@ object JsonResponseObjects {
             case RenderInheritedProperties.JSON =>
               JRPropertyHierarchyJson(list.map(JRParentProperty.fromParentProperty(_)))
           }
-          (Some(parents), prop.hierarchy.headOption.map(_.value))
+          Some(parents)
       }
-      val hierarchyStatus    = JRPropertyHierarchyStatus.fromParentProperties(hasChildTypeConflicts, fullHierarchy)
-      val desc               = if (prop.prop.description.trim.isEmpty) None else Some(prop.prop.description)
-      JRProperty(
-        prop.prop.name,
-        prop.prop.value,
-        desc,
-        prop.prop.inheritMode,
-        prop.prop.provider,
-        parents,
-        hierarchyStatus,
-        origval
-      )
     }
+
+    /**
+     * The first element of the hierarchy in the original order is the original value
+     * FIXME: there is an ordering, we could enforce a specific sorted hierarchy 
```

Comment:
you should not put fixme in the code. They never get corrected. 
What is the cheapest change you can make to help make that FIXME useless ? 
Creating a "parents" datastructure (ie ~ a list) is not hard, but we don't really want to maintain one more of these and pay the conversion price all around. 
An opaque type would be ideal, but we don't have them yet. 
Adding a comment on all methods saying "direct parent is first, root is last" (or the opposite) cheap. 
Completed with some unit tests to enforce it there may be enough for now, and better than a fixme. 
Perhaps there is other simple solution, but I don't see them now.

Reply:
okay, I thought about `SortedSet` in _cats_ but using them everywhere in our API would be too much of change so I guess when opaque types will be easier to use we'll add it.

So should I just remove the "FIXME:" portion and keep the suggestion there in the comment with a big "IF" ? :grin: